agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -84,7 +84,7 @@ def ensure_desired_state
       elsif service_container && service_pod.running? && !service_container.running?
         info "starting #{service_pod.name}"
         ensure_started
-      elsif service_pod.running? && (service_container && service_container_outdated?(service_container, service_pod))
+      elsif service_pod.running? && (service_container && service_container_outdated?(service_container))
```

Comment:
Might be worth fixing #2154 too: the agent can hit the earlier `!service_container.running?` -> `ensure_started` case even if the container is outdated.

Plus, right now it looks like the `recreate_service_container?` check is broken and will never be hit, because the agent will always just start the `!running?` container instead of re-creating it?